File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/reservations/model/event/ReservationDeleted.java
package de.vkb.dojo.es.reservations.model.event;

import com.fasterxml.jackson.annotation.*;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("reservatipn-deleted")
public class ReservationDeleted extends ReservationEvent {
    private final String operationId;
    private final String aggregateId;

    @JsonCreator
    public ReservationDeleted(
            @JsonProperty("operationId") String operationId,
            @JsonProperty("aggregateId") String aggregateId
    ) {
        this.operationId = operationId;
        this.aggregateId = aggregateId;
    }

    @Override
    @JsonGetter
    public String getOperationId() {
        return operationId;
    }


    @Override
    @JsonGetter
    public String getAggregateId() {
        return aggregateId;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/reservations/model/event/ReservationDeleted.kt
package de.vkb.dojo.es.reservations.model.event

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("reservation-deleted")
data class ReservationDeleted(
    override val operationId: String,
    override val aggregateId: String
): ReservationEvent
